arch/aarch64/crc32_acle.c

```diff
@@ -15,66 +15,80 @@
 #ifdef __linux__
 #  include <stddef.h>
 #endif
+#include "zutil.h"
 
 uint32_t crc32_acle(uint32_t crc, const unsigned char *buf, uint64_t len) {
-    register uint32_t c;
-    register const uint16_t *buf2;
-    register const uint32_t *buf4;
-    register const uint64_t *buf8;
+    uint32_t c = ~crc;
 
-    c = ~crc;
     if (len && ((ptrdiff_t)buf & 1)) {
         c = __crc32b(c, *buf++);
         len--;
     }
 
-    if ((len > 2) && ((ptrdiff_t)buf & 2)) {
-        buf2 = (const uint16_t *) buf;
-        c = __crc32h(c, *buf2++);
-        len -= 2;
-        buf4 = (const uint32_t *) buf2;
-    } else {
-        buf4 = (const uint32_t *) buf;
+    if ((len > sizeof(uint16_t)) && ((ptrdiff_t)buf & 2)) {
+        uint16_t val;
+        MEMCPY(&val, buf, sizeof(val));
```

Comment:
This seems unnecessary as buf is guaranteed to be aligned...

Reply:
This and the other comments below are now fixed in the new patch.
The new patch minimizes the number of changes: it keeps all the auxiliary pointers buf2, buf4, and buf8.